### PR TITLE
Load Reporting Core new features

### PR DESCRIPTION
**Pull request by dgquintas**
- Propagation of (rpc) method name.
- Invocation of the hook at (call, channel) x (creation, destruction)
- Added enum to identify the source of invocation.
- Fixed testing. Went from test fixture to simple test.
- Correct locking of LR callback.

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/writing.c [hunk 1]
```diff
@@ -262,6 +263,10 @@ static void finalize_outbuf(grpc_exec_ctx *exec_ctx,
             stream_writing->id, &stream_writing->flow_controlled_buffer,
             send_bytes, is_last_frame, &stream_writing->stats,
             &transport_writing->outbuf);
+        if (is_first_data_frame) {
+          stream_writing->stats.data_bytes -= 5; /* discount grpc framing */
```
Comment (dgquintas) [inline src/core/ext/transport/chttp2/transport/writing.c]: the 5 could be made "better" behind a #define, at least...

File: src/core/ext/load_reporting/load_reporting.h [hunk 2]
```diff
@@ -34,19 +34,51 @@
...
+  GRPC_LR_POINT_CALL_DESTRUCTION
+} grpc_load_reporting_source;
+
+/** Call information to be passed to the provided LR callback. */
 typedef struct grpc_load_reporting_call_data {
-  const grpc_call_stats *stats;   /**< Stats for the call */
-  const char *trailing_md_string; /**< LR trailing metadata info */
+  const grpc_load_reporting_source source; /**< point of last data update. */
+
+  // XXX
```
Comment (thinkerou) [inline src/core/ext/load_reporting/load_reporting.h]: what's XXX mean?

Comment (dgquintas) [inline src/core/ext/load_reporting/load_reporting.h]: it's a mark I put to remember to do something, in this case add more comments. This PR is still not ready for review (but almost), that's why it doesn't have anyone assigned to it :)

**Comment by dgquintas** [comment]
This is going back to the drawing board. Closing for now.

**Pull request closed without merging**